File: clojure/src/stats/hand_shakes.clj
(ns stats.hand_shakes
  (:gen-class)
  (:require [clojure.math.combinatorics :as combo]))

(defn combination-count [number_of_people]
  (count (combo/combinations (range 0 number_of_people) 2)))

(defn usage []
  "Usage: 
    lein -m stats.hand_shakes [number_of_people]")

(defn -main
  "How many unique hand shakes with n people"
  [& args]
  ;; work around dangerous default behaviour in Clojure
  (alter-var-root #'*read-eval* (constantly false))

  (if (= 1 (count args))
    (let [number_of_people (read-string (first args))]
      (println (combination-count number_of_people)))
    (println (usage))))

File: ruby/hand_shakes.rb
#!/usr/bin/env ruby

class HandShakes
  def initialize(options={})
    @people = (1..options[:people].to_i).to_a
  end

  def combinations
    @people.combination(2)
  end
end

if __FILE__ == $0
  if ARGV.count == 1
    @hand_shakes = HandShakes.new :people => ARGV.first
    puts @hand_shakes.combinations.count
  else
    puts "Usage: \n\t./hand_shakes.rb [number_of_people]"
  end
end
